src/tests/test_magic.py

```diff
@@ -187,6 +187,16 @@ def test_persist_no_index(ip):
     assert persisted == [(1, "foo"), (2, "bar")]
 
 
+def test_unrecognized_arguments_cell_magic(ip):
+    result = ip.run_cell("%%sql --stuff \n SELECT * FROM test")
+    assert "Unrecognized argument(s)" in str(result.error_in_exec)
+
+
+def test_unrecognized_arguments_line_magic(ip):
```

Comment:
we *should not* validate arguments in the line magic. this PR should only apply to the cell magic (`%%sql`)

Reply:
Remove this test and update now